packages/react-components/src/AccessibilityScenarios/Input.stories.tsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react';
+
+import { Input, InputOnChangeData } from '@fluentui/react-input';
+import { Label } from '@fluentui/react-label';
+import { Button } from '@fluentui/react-button';
+
+import { Scenario } from './utils';
+
+export const RegistrationFormInputsAccessibilityScenario = () => {
+  // eslint-disable-next-line
+  const validations: Record<string, any> = {
+    fullName: {
+      type: 'field',
+      regexes: [/^[A-Za-zÀ-ÖØ-öø-ÿěščřžďťňůĚŠČŘŽĎŤŇŮ -]{2,50}$/],
+    },
+    nickname: {
+      type: 'field',
+      regexes: [/^[A-Za-zÀ-ÖØ-öø-ÿěščřžďťňůĚŠČŘŽĎŤŇŮ -]{2,20}$|^$/],
+    },
+    password: {
+      type: 'password',
+      regexes: [
+        /^[^ ]{8,30}$/,
+        /^[^ ]*[0-9][^ ]*$/,
+        /^[^ ]*[A-Z][^ ]*$/,
+        /^[^ ]*[a-z][^ ]*$/,
+        /^[^ ]*[^A-Za-zÀ-ÖØ-öø-ÿěščřžďťňůĚŠČŘŽĎŤŇŮ ][^ ]*$/,
+      ],
+    },
+  };
+
+  const [values, setValues] = React.useState<Record<string, string>>({
+    fullName: '',
+    nickname: '',
+    password: '',
+  });
+  const [invalids, setInvalids] = React.useState<Record<string, boolean>>({
+    fullName: false,
+    nikname: false,
+    password: false,
+  });
+  const [isErrorMessageVisible, setIsErrorMessageVisible] = React.useState<Record<string, boolean>>({
+    fullName: false,
+    nickname: false,
+    password: false,
+  });
+  const [isPasswordVisible, setIsPasswordVisible] = React.useState(false);
+  const [isSubmitted, setIsSubmitted] = React.useState(false);
+
+  const handleControlChange = (event: React.FormEvent, data: InputOnChangeData, name: string) => {
+    const stateUpdate = { [name]: data.value };
+    setValues({ ...values, ...stateUpdate });
+  };
+  const handleShowPasswordChange = (event: React.ChangeEvent) => {
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+  const handleControlBlur = (event: React.FormEvent, name: string) => {
+    validateControl(name);
+  };
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
```

Comment:
Shouldn't you only prevent the default if the form validation failed? In a real case, someone might want the default submission behavior.

Reply:
This is a single page application, so it doesn't send the form data neither retrieves the data from the server. Therefore the form submission must be prevented.